### ISSUE TITLE
state of nats/stan on k8s?

### ISSUE THREAD

**Issue by justinrush** [nats-io/k8s#125]
tl/dr: I'm interested in the vision and near/medium term future of how one deploys NATS/STAN on k8s, it feels like there is development time being spent on a variety of different and conflicting ways with sparse documentation. I have no particular preference of operator over helm, etc, but would prefer to spend my time writing code rather than fighting with getting it deployed.

Have been using rabbitMQ, poked around a bit at the NATS libraries and love how easy it is to interact with nats using the golang library. Then I tried to deploy stan and went down a rabbit hole of hell trying to figure out how to have nats/stan with a user/pass, persistent volume, and externally available. There seem to be the following ways to deploy NATS on k8s:
* An all in one script that has very sparse documentation and doesn't allow overrides for things as simple as namespace. It also doesn't work anymore (tested on a fresh rke deployed cluster).
* Helm chart from bitnami that seems to work quite well and be straight forward
* Helm chart from nats that may work but is not straight forward
* Operator that makes things easy, but has sparse documentation. Still have no idea how to do persistent volumes and I have to manually create a service of type loadbalancer afterwards to get accessible outside of the cluster.

Getting STAN deployed was rough. The operator seems to be the favored approach but there are a bunch of bugs that people are just working around - like making sure you have "config: {}" in the spec otherwise you can't deploy multi-node cluster.  It doesn't support service account auth and there is no documentation on how to configure user/pass auth to nats outside of an issue where someone posted a workaround.

**Comment by wallyqs** [comment]
Hi @justinrush thanks for the feedback, and apologies for the confusion. Going forward the helm charts in this repo would hope that they are easier to get started, for deploying STAN the info could be found in the readme here: https://github.com/nats-io/k8s/tree/master/helm/charts/stan#tldr 

We don't have an example of using L7 load balancers in this repo but you can find some in the docs here for AWS https://docs.nats.io/nats-on-kubernetes/nats-external-nlb
One issue with load balancers is that unless you do TLS termination at the load balancer, you wouldn't be able to have secure connection to NATS due to how the protocol works.  L4 load balancers should work ok though like in the AWS example.

Btw the one line installer can fail in some environments when there is no public ip provision since it relies on using host ports for external access to not use load balancers, maybe that is what happened in the RKE cluster.

**Comment by justinrush** [comment]
Thanks, thats helpful to know the direction
